### ISSUE TITLE
LSTM export ONNX:  Non-zero status code returned while running ScatterElements node. Name:'ScatterElements_18' Status Message: Indices vs updates dimensions differs at position=0 11 vs 8

### ISSUE THREAD

**Issue by wangkenpu** [microsoft/onnxruntime#11830]
**Describe the bug**
I dump a LSTM PyTorch model to the ONNX model, but facing an error `onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running ScatterElements node. Name:'ScatterElements_18' Status Message: Indices vs updates dimensions differs at position=0 11 vs 8` even I set the `batch_size` as the dynamic axis.

**System information**
- Docker image: `wangkenpu/assessment/pytorch1.10.0-py38-cuda11.1-cudnn8-ubuntu18.04`

**To Reproduce**

```python
class ToyModel(nn.Module):

    def __init__(self):
        super().__init__()
        self.rnn_num_layers = 2
        self.rnn_hidden_size = 64
        self.rnn = nn.LSTM(input_size=10, hidden_size=self.rnn_hidden_size, num_layers=self.rnn_num_layers,
                           batch_first=True)

    def forward(self, x, length):
        length = length.reshape(-1).cpu()
        batch_size = x.size(0)
        # Pack padded sequence
        x = nn.utils.rnn.pack_padded_sequence(x, length, batch_first=True, enforce_sorted=False)
        h0 = torch.zeros(self.rnn_num_layers, batch_size, self.rnn_hidden_size)
        c0 = torch.zeros(self.rnn_num_layers, batch_size, self.rnn_hidden_size)
        out, _ = self.rnn(x, (h0, c0))
        # Unpack the output
        out, _ = nn.utils.rnn.pad_packed_sequence(out, batch_first=True)
        return out


model = ToyModel()

dummy_input = (torch.randn(8, 40, 10).float(), torch.randint(20, 40, (8, 1)).long())
onnx_model_path = "./model.onnx"
dynamic_axes = {
    "input": {0: "batch_size", 1: "seq_length"},
    "length": {0: "batch_size"},
    "output": {0: "batch_size"}
}
input_names = ["input", "length"]
output_name = ["output"]
torch.onnx.export(model, dummy_input, onnx_model_path, input_names=input_names, output_names=output_name,
                  dynamic_axes=dynamic_axes, opset_version=14)
ort_input = {}
batch_size = 11
seq_length = 100
ort_input["input"] = torch.randn(size=(batch_size, seq_length, 10)).numpy()
ort_input["length"] = torch.randint(low=5, high=seq_length + 1, size=(batch_size, 1)).numpy()
ort_session = ort.InferenceSession(onnx_model_path)
output = ort_session.run(["output"], ort_input)
```

Running the code using the docker image I mentioned above.

**Screenshots**
![image](https://user-images.githubusercontent.com/15028609/173304622-b14e72df-90b6-4514-89d8-c17b34d50a19.png)

**Comment by wangkenpu** [comment]
When I do not use `nn.utils.rnn.pack_padded_sequence` and `nn.utils.rnn.pad_packed_sequence`, the behavior is expected.

**Comment by garymm** [comment]
Can you try with the latest PyTorch nightly and ORT nightly packages?

https://pytorch.org/get-started/locally/
https://test.pypi.org/project/ort-nightly/

**Comment by wangkenpu** [comment]
Thanks, @garymm. It can solve the issue and just report some `UserWarning`.

```shell
tools/miniconda/envs/onnx-nightly/lib/python3.8/site-packages/torch/onnx/_patch_torch.py:67: UserWarning: The shape inference of prim::PackPadded type is missing, so it may result in wrong shape
inference for the exported graph. Please consider adding it in symbolic function. (Triggered internally at  /opt/conda/conda-bld/pytorch_1655709061066/work/torch/csrc/jit/passes/onnx/shape_type_inference.cpp:1898.)
  torch._C._jit_pass_onnx_node_shape_type_inference(
tools/miniconda/envs/onnx-nightly/lib/python3.8/site-packages/torch/onnx/symbolic_opset9.py:3485: UserWarning: Exporting a model to ONNX with a batch_size other than 1, with a variable length with LSTM can cause an error when running the ONNX model with a different batch size. Make sure to save the model with a batch size of 1, or define the initial states (h0/c0) as inputs of the model.
  warnings.warn(
tools/miniconda/envs/onnx-nightly/lib/python3.8/site-packages/torch/onnx/_patch_torch.py:67: UserWarning: The shape inference of prim::PadPacked type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function. (Triggered internally at  /opt/conda/conda-bld/pytorch_1655709061066/work/torch/csrc/jit/passes/onnx/shape_type_inference.cpp:1898.)
  torch._C._jit_pass_onnx_node_shape_type_inference(
```
I'd like to know do we have some plan to release it as the stable version?
